pkg/traceql/engine_metrics_compare.go

```diff
@@ -242,10 +243,15 @@ func (m *MetricsCompare) result(multiplier float64) SeriesSet {
 			}
 
 			top.get(m.topN, func(v Static) {
+				value = values[v.MapKey()]
```

Comment:
Can we dig into understanding this a bit more?  How can a value be in the topN but not in the map? Since we just populated topN from `values` in the previous loop.